cmd/client_test.go

```diff
@@ -86,139 +89,113 @@ func generateRedisProbeDef(name string) *probes_configpb.ProbeDef {
 }
 
 // TestRegisterAndAddProbe tests that an extension probe type can be registered and added to Cloudprober without error.
-func TestRegisterAndAddProbe(t *testing.T) {
-	cpc := &CloudproberClient{}
-	cpc.InitClient(rpcServer)
-
-	// Create a probe and then register and add it to the prober
-	err := cpc.RegisterAndAddProbe(int(myprobe.E_RedisProbe.Field), generateRedisProbeDef(testProbe0), &myprobe.Probe{})
-	if err != nil {
-		t.Errorf("Probe not correctly registered and added to Cloudprober, error: %v", err)
-	}
-
-	// Check if the probe was added correctly
-	resp, errResp := cpc.ListProbes()
-	if errResp != nil {
-		t.Errorf("ListProbes() failed, error: %v", errResp)
-	}
-
-	respProbes := resp.GetProbe()
-
-	if respProbes[0].GetName() != testProbe0 {
-		t.Errorf("Expected probe %s to be running, got probe %s running", testProbe0, respProbes[0].GetName())
-	}
-
-	// Try adding a second probe to see if the registration is still in place
-	err = cpc.RegisterAndAddProbe(int(myprobe.E_RedisProbe.Field), generateRedisProbeDef(testProbe1), &myprobe.Probe{})
-
-	// Check if the probe was added correctly
-	resp, errResp = cpc.ListProbes()
-	if errResp != nil {
-		t.Errorf("ListProbes() failed, error: %v", errResp)
-	}
-
-	respProbes = resp.GetProbe()
-
-	if respProbes[1].GetName() != testProbe1 {
-		t.Errorf("Expected probe %s to be running, got probe %s running", testProbe0, respProbes[1].GetName())
-	}
-
-	err = cpc.RemoveProbe(testProbe0)
-	if err != nil {
-		t.Errorf("RemoveProbe() failed, error: %v", err)
-	}
-
-	err = cpc.RemoveProbe(testProbe1)
-	if err != nil {
-		t.Errorf("RemoveProbe() failed, error: %v", err)
-	}
-
-	err = cpc.CloseConn()
-	if err != nil {
-		t.Errorf("CloseConn() failed, error: %v", err)
+func testRegisterAndAddProbe(t *testing.T) {
+	// Setup Cloudprober and gRPC client connection
+	testHermes, client := setupCloudproberAndClient()
+	defer teardownCloudproberAndClient(testHermes, client)
+
+	for i := 0; i < len(testProbes); i++ {
```

Comment:
Ideally you should declare `testProbes` closer to this.

Reply:
I have put in a TODO to revamp the test structure to make it true table-driven testing.